go/vt/mysqlctl/s3backupstorage/s3.go

```diff
@@ -110,6 +111,23 @@ var logNameMap logNameToLogLevel
 
 const sseCustomerPrefix = "sse_c:"
 
+type endpointResolver struct {
+	r        s3.EndpointResolverV2
+	endpoint *string
+}
+
+func (er *endpointResolver) ResolveEndpoint(ctx context.Context, params s3.EndpointParameters) (transport.Endpoint, error) {
+	params.Endpoint = er.endpoint
+	return er.r.ResolveEndpoint(ctx, params)
+}
+
+func newEndpointResolver() *endpointResolver {
+	return &endpointResolver{
+		r:        s3.NewDefaultEndpointResolverV2(),
+		endpoint: &endpoint,
```

Comment:
Would be nice IMO to have a comment about what this is here: 
```suggestion
		endpoint: &endpoint, // Use default endpoint of amazonaws.com
```

Reply:
this is already the case in the variable definition:
```
// AWS endpoint, defaults to amazonaws.com but appliances may use a different location
	endpoint string
```